src/Ng/Program.cs

```diff
@@ -115,6 +128,40 @@ public static async Task MainAsync(string[] args)
             applicationInsightsConfiguration?.TelemetryConfiguration.TelemetryChannel.Flush();
         }
 
+        /// <summary>
+        /// This mimics the approach taking in NuGet.Job's JsonConfigurationJob. We add common infrastructure to the
+        /// dependency injection container and load Autofac modules in the current assembly. This gives us a hook to
+        /// customize the initialization of this program. Today this service provider is only used for modifying
+        /// telemetry configuration but it could eventually be used in the <see cref="NgJobFactory"/>.
+        /// </summary>
+        private static IServiceProvider InitializeServiceProvider(
+            IDictionary<string, string> arguments,
+            ISecretInjector secretInjector,
+            ApplicationInsightsConfiguration applicationInsightsConfiguration,
+            ITelemetryClient telemetryClient,
+            ILoggerFactory loggerFactory)
+        {
+            var configurationBuilder = new ConfigurationBuilder()
+                .AddInMemoryCollection(arguments);
+
+            IServiceCollection services = new ServiceCollection();
+            services.AddSingleton(secretInjector);
+            services.AddSingleton(applicationInsightsConfiguration.TelemetryConfiguration);
+            services.AddSingleton<IConfiguration>(configurationBuilder.Build());
+
+            services.Add(ServiceDescriptor.Scoped(typeof(IOptionsSnapshot<>), typeof(NonCachingOptionsSnapshot<>)));
+            services.AddSingleton(loggerFactory);
+            services.AddLogging();
+
+            services.AddSingleton(telemetryClient);
+
+            var containerBuilder = new ContainerBuilder();
+            containerBuilder.Populate(services);
+            containerBuilder.RegisterAssemblyModules(typeof(Program).Assembly);
```

Comment:
It was too much to move Ng to `JsonConfigurationJob`. This enables the same thing that we need though: loading Autofac modules in the same assembly.